File: code.py
def luhn_check(card_number: str) -> bool:
    total = 0
    double = False

    for i in range(len(card_number) - 1, -1, -1):
        d = int(card_number[i])

        if double:
            d *= 2
            if d > 9:
                d -= 9

        total += d
        double = not double

    return total % 10 == 0


# Example usage:
print(luhn_check("4532015112830366"))  # True

File: code.js
function luhnCheck(cardNumber) {
    let sum = 0;
    let doubleDigit = false;

    for (let i = cardNumber.length - 1; i >= 0; i--) {
        let d = parseInt(cardNumber[i]);

        if (doubleDigit) {
            d *= 2;
            if (d > 9) d -= 9;
        }

        sum += d;
        doubleDigit = !doubleDigit;
    }

    return sum % 10 === 0;
}

// Example usage:
console.log(luhnCheck("4532015112830366")); // true (valid)
